src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.cs

```diff
@@ -1311,8 +1311,183 @@ public static Matrix4x4 CreateReflection(Plane value)
+                Vector128<float> C2 = Sse.Multiply(V01, V11);
+                Vector128<float> C4 = Sse.Multiply(V02, V12);
+                Vector128<float> C6 = Sse.Multiply(V03, V13);
+
+                // V11 = D0X,D0Y,D2X,D2X
+                V11 = Sse.Shuffle(D0, D2, 0x4);
+                V00 = Sse.Shuffle(row2, row2, 0x9e);
+                V10 = Sse.Shuffle(D0, V11, 0x93);
+                V01 = Sse.Shuffle(row1, row1, 0x7b);
+                V11 = Sse.Shuffle(D0, V11, 0x26);
+
+                // V13 = D1X,D1Y,D2Z,D2Z
+                V13 = Sse.Shuffle(D1, D2, 0xa4);
+                V02 = Sse.Shuffle(row4, row4, 0x9e);
+                V12 = Sse.Shuffle(D1, V13, 0x93);
+                V03 = Sse.Shuffle(row3, row3, 0x7b);
+                V13 = Sse.Shuffle(D1, V13, 0x26);
+
+                C0 = Sse.Subtract(C0, Sse.Multiply(V00, V10));
+                C2 = Sse.Subtract(C2, Sse.Multiply(V01, V11));
+                C4 = Sse.Subtract(C4, Sse.Multiply(V02, V12));
+                C6 = Sse.Subtract(C6, Sse.Multiply(V03, V13));
+                V00 = Sse.Shuffle(row2, row2, 0x33);
+
+                // V10 = D0Z,D0Z,D2X,D2Y
+                V10 = Sse.Shuffle(D0, D2, 0x4A);
+                V10 = Sse.Shuffle(V10, V10, 0x2C);
+                V01 = Sse.Shuffle(row1, row1, 0x8D);
+
+                // V11 = D0X,D0W,D2X,D2Y
+                V11 = Sse.Shuffle(D0, D2, 0x4C);
+                V11 = Sse.Shuffle(V11, V11, 0x93);
+                V02 = Sse.Shuffle(row4, row4, 0x33);
+
+                // V12 = D1Z,D1Z,D2Z,D2W
+                V12 = Sse.Shuffle(D1, D2, 0xEA);
+                V12 = Sse.Shuffle(V12, V12, 0x2C);
+                V03 = Sse.Shuffle(row3, row3, 0x8D);
+
+                // V13 = D1X,D1W,D2Z,D2W
+                V13 = Sse.Shuffle(D1, D2, 0xEC);
+                V13 = Sse.Shuffle(V13, V13, 0x93);
+
+                V00 = Sse.Multiply(V00, V10);
+                V01 = Sse.Multiply(V01, V11);
+                V02 = Sse.Multiply(V02, V12);
+                V03 = Sse.Multiply(V03, V13);
+
+                Vector128<float> C1 = Sse.Subtract(C0, V00);
+                C0 = Sse.Add(C0, V00);
+                Vector128<float> C3 = Sse.Add(C2, V01);
+                C2 = Sse.Subtract(C2, V01);
+                Vector128<float> C5 = Sse.Subtract(C4, V02);
+                C4 = Sse.Add(C4, V02);
+                Vector128<float> C7 = Sse.Add(C6, V03);
+                C6 = Sse.Subtract(C6, V03);
+
+                C0 = Sse.Shuffle(C0, C1, 0xD8);
+                C2 = Sse.Shuffle(C2, C3, 0xD8);
+                C4 = Sse.Shuffle(C4, C5, 0xD8);
+                C6 = Sse.Shuffle(C6, C7, 0xD8);
+
+                C0 = Sse.Shuffle(C0, C0, 0xD8);
+                C2 = Sse.Shuffle(C2, C2, 0xD8);
+                C4 = Sse.Shuffle(C4, C4, 0xD8);
+                C6 = Sse.Shuffle(C6, C6, 0xD8);
+
+                // Get the determinant
+                vTemp2 = row1;
+                Vector128<float> vTemp = Sse.Multiply(C0, vTemp2);
```

Comment:
You should be able to use `Vector4.Dot(C0.AsVector4(), vTemp2.AsVector4()).AsVector128()` so we can use `DPPS` on hardware with it available. `AsVector4` and `AsVector128` should be zero cost conversions.

Reply:
The result of Vector4.Dot is a float,  I used that to pack a Vector128<Float>.Create for use later in the method.  Also updated the Determinant check to use only the float from the Dot method.